SunEngine.Cli/StartupConfiguration.cs

```diff
@@ -68,20 +72,41 @@ private string GetConfigurationDirectory(IEnumerable<string> arguments)
             var configurationProperty = arguments.FirstOrDefault(x => x.StartsWith(ConfigurationArgumentName));
             if (string.IsNullOrEmpty(configurationProperty))
             {
-                Logger.Warn("Property for configuration wasn't set. Default configuration will be used.");
+                Console.WriteLine("Property for configuration wasn't set. Default configuration will be used.");
                 return DefaultConfigurationFileName;
             }
 
             var configurationFileName = configurationProperty.Substring(ConfigurationArgumentName.Length).Trim();
             if (string.IsNullOrEmpty(configurationFileName))
             {
-                Logger.Warn("Property for configuration was empty or blank. Default configuration will be used.");
+                Console.WriteLine("Property for configuration was empty or blank. Default configuration will be used.");
                 return DefaultConfigurationFileName;
             }
 
-            Logger.Info($"Configuration file {configurationFileName} will be used.");
+            if (!configurationFileName.Equals(DefaultConfigurationFileName) && !configurationFileName.EndsWith(ConfigurationDirectoryNameEnd))
+            {
+                configurationFileName += ConfigurationDirectoryNameEnd;
+            }
+
+            Console.WriteLine($"Configuration directory \"{configurationFileName}\" will be used.");
             return configurationFileName;
         }
+
+        public void ExitApplicationIfConfigurationIsNotValid()
+        {
+            bool failed = !TestIfConfigurationDirectoryExists(ConfigurationDirectoryRoute);
+
+            if(failed)
+                Environment.Exit(0);
```

Comment:
This class stores configuration, why we use Environment.Exit(0); in it?
Environment.Exit(0); should be used in Program class, not here. It's an abstraction violation.

Try to add unit test for this method.

Reply:
I refactor this. Check now.